tests/routes/synapse.rs

```diff
@@ -0,0 +1,171 @@
+#[cfg(test)]
+mod synapse_sniff {
+
+    mod login {
+        use std::collections::HashMap;
+
+        use crate::helpers::server::{create_test_db, get_configuration};
+        use actix_web::{test, web::Data};
+        use faux::when;
+        use social_service::{
+            components::{
+                app::{AppComponents, CustomComponents},
+                synapse::{
+                    AuthChain, LoginIdentifier, SynapseComponent, SynapseLoginRequest,
+                    SynapseLoginResponse,
+                },
+            },
+            get_app_router,
+            routes::v1::error::CommonError,
+        };
+
+        const URL: &str = "/_matrix/client/r0/login";
+
+        #[actix_web::test]
+        async fn should_be_200_and_has_user_in_cache() {
+            let config = get_configuration();
+
+            // Just mocks synapse
+            let mut mocked_synapse = SynapseComponent::faux();
+
+            when!(mocked_synapse.login).then(|_| {
+                let mut mocked_hash_map = HashMap::new();
+                let mut mocked_hash_map_2 = HashMap::new();
+                mocked_hash_map_2.insert(
+                    "base_url".to_string(),
+                    "https://synapse.decentraland.zone".to_string(),
+                );
+                mocked_hash_map.insert("m.homeserver".to_string(), mocked_hash_map_2);
+                Ok(SynapseLoginResponse {
+                    user_id: "0xA1".to_string(),
+                    access_token: "0xA1_TOKEN".to_string(),
+                    device_id: "0xA1_DEVICE".to_string(),
+                    home_server: "decentraland.zone".to_string(),
+                    well_known: mocked_hash_map,
+                })
+            });
+
+            let mocked_comps = CustomComponents {
+                synapse: Some(mocked_synapse),
+                db: None,
+                users_cache: None,
+                redis: None,
+            };
```

Comment:
How about adding a builder for CustomComponents and have sth like `CustomComponents::builder().with_synapse(mock_synapse).build()` here?

Reply:
cool, added!